test/cpp/asyncprogressworkersignal.cpp

```diff
@@ -0,0 +1,66 @@
+/*********************************************************************
+ * NAN - Native Abstractions for Node.js
+ *
+ * Copyright (c) 2016 NAN contributors
+ *
+ * MIT License <https://github.com/nodejs/nan/blob/master/LICENSE.md>
+ ********************************************************************/
+
+#ifndef _WIN32
+#include <unistd.h>
+#define Sleep(x) usleep((x)*1000)
+#endif
+#include <nan.h>
+
+using namespace Nan;  // NOLINT(build/namespaces)
+
+class ProgressWorker : public AsyncProgressWorker {
+ public:
+  ProgressWorker(
+      Callback *callback
+    , Callback *progress
+    , int milliseconds
+    , int iters)
+    : AsyncProgressWorker(callback), progress(progress)
+    , milliseconds(milliseconds), iters(iters) {}
+  ~ProgressWorker() {}
+
+  void Execute (const AsyncProgressWorker::ExecutionProgress& progress) {
+    for (int i = 0; i < iters; ++i) {
+      progress.Signal();
+      Sleep(milliseconds);
+    }
+  }
+
+  void HandleProgressCallback(const char *data, size_t size) {
+    HandleScope scope;
+
+    v8::Local<v8::Value> argv[] = {
+        New<v8::Boolean>(data == NULL && size == 0)
+    };
+    progress->Call(1, argv);
```

Comment:
This code is fine as is but there's a shorthand version:
```c++
v8::Local<v8::Value> arg = New<v8::Boolean>(data == NULL && size == 0);
progress->Call(1, &arg);
```
Feel free to ignore. :-)